CanvasColorSpaceProposal.md

```diff
@@ -26,34 +27,74 @@ Some implementations convert images drawn to canvases to the sRGB color space. T
 
 * <cite>[https://github.com/whatwg/html/issues/299]</cite> <blockquote><p>Allow 2dcontexts to use deeper color buffers</p></blockquote>
 * <cite>[https://bugs.chromium.org/p/chromium/issues/detail?id=425935]</cite> <blockquote><p>Wrong color profile with 2D canvas</p></blockquote>
-* Engineers from the Google Photos, Maps and Sheets teams have expressed a desire for canvases to become color managed.  Particularly for the use case of resizing an imaging, using a canvas, prior to uploading it to the server, to save bandwidth. The problem is that the images retrieved from a canvas are in an undefined color space and no color space information is encoded by toDataURL or toBlob. 
+* Engineers from the Google Photos, Maps and Sheets teams have expressed a desire for canvases to become color managed.  Particularly for the use case of resizing an imaging, using a canvas, prior to uploading it to the server, to save bandwidth. The problem is that the images retrieved from a canvas are in an undefined color space and no color space information is encoded by toDataURL or toBlob.
 
 ## Proposed Solution
 
 * Clearly define the color space of canvases.  Most current browser implementations are either color-managed or are currently actively working on becoming color-managed.  Therefore, it will be possible to specify that the default color space of canvases as 'srgb' instead of leaving it undefined in the spec (as is currently the case due to lack of interoperability).
 * Add a canvas context creation attribute to specify a color space.
-* Add a canvas context creation attribute to specify a numeric format for storing pixel values.
-* Color space and numeric format parameters are also extended to image storage interfaces that interact with canvases, such as CanvasPattern, ImageData and ImageBitmap.
+* Add a canvas context creation attribute to specify an encoding format for storing pixel values.
+* Add a canvas context creation attribute to specify a linearity for perceptual vs physical blending/interpolation.
+* Color space, encoding format, and linearity parameters are also extended to image storage interfaces that interact with canvases, such as CanvasPattern, ImageData and ImageBitmap.
 
 ### Processing Model
 
-#### The colorSpace canvas creation parameter
-
 IDL:
 <pre>
-enum CanvasColorSpace {
+// Feature enums:
+
+enum CanvasColorSpaceEnum {
   "srgb", // default
-  "linear-srgb",
+  "rec-2020",
```

Comment:
@ccameron-chromium and @FredrikHubinette have worked extensively on Chrome's color profile support and would ideally comment on this change. There were some thoughts that "extended-srgb" would be a more portable choice for a color space supporting wide-color gamut and HDR use cases. Agree that the previous iteration of this spec seemed to have accidentally dropped the rec-2020 enum without fully removing references to it.

Reply:
I think this is supported by this PR. (srgb+float16)
What I don't think we want is the srgb-chromaticity+rec2020-trans-func combo from the previous version. That would be an unfortunate subtle deviation from the CSS colors spec.

Reply:
I would include the full list from the color-gamut media query
https://developer.mozilla.org/en-US/docs/Web/CSS/@media/color-gamut
So that would be srgb, p3, and rec2020.

I think we should also include extended-srgb and linear-extended-srgb as options.